### ISSUE TITLE
HEIF & HEVC  - iOS 11

### ISSUE THREAD

**Issue by OliverDimitrov** [zhangao0086/DKImagePickerController#378]
Hello @zhangao0086 ,
With iOS 11, Apple introduced the new image compression (HEIF) and video encoding (HEVC). 
Will this cause any problems with the existing implementation? 

Currently I'm developing an app using this library, but I don't own device capable for HEIF and HEVC so I'm unable to test it. Can you shine some light on this matter? 

Thanks.

**Comment by zhangao0086** [comment]
Hi @OliverDimitrov  I'll test it in the next few days.

**Comment by zhangao0086** [comment]
I was tested with iPhone 7 Plus.

#### Compatibility

If you're using `fetch UIImage` related APIs, such as `fetchImageWithSize...` then you won't have compatibility issues. You have to convert UIImage to Data with the following two functions:
```swift
public func UIImagePNGRepresentation(_ image: UIImage) -> Data?
public func UIImageJPEGRepresentation(_ image: UIImage, _ compressionQuality: CGFloat) -> Data?
```
There is no `UIImageHEIFRepresentation`, so the UIImage will transcode HEIF to JPEG or PNG.

If you're using `fetchImageDataForAsset` to get the original data of the image, then you may encounter compatibility issues. In future version, I'll do:
* Returns `dataUTI` in `completeBlock`, so you can check the `dataUTI` with `AVFileTypeHEIC` or `AVFileTypeHEIF` or `...`
* I'll add an option to determine whether automatically transcode HEIF to JPEG, just like the newly property of the `UIImagePickerController`:
```swift
@available(iOS 11.0, *)
open var imageExportPreset: UIImagePickerControllerImageURLExportPreset // default value is UIImagePickerControllerImageExportPresetCompatible.
```

**Comment by zhangao0086** [comment]
#370
